client/src/pages/client/Services.tsx

```diff
@@ -27,11 +32,14 @@ const Services = (props: Props): React.JSX.Element => {
     address: "",
     note: "",
     service: "",
+    isRegister: false,
+    password: "",
   };
   const [isLoading, setIsLoading] = useState(false);
   const [services, setServices] = useState<StateProps[]>([]);
   const [showModal, setShowModal] = useState(false);
```

Comment:
instead of showModal we could make it like this
```suggestion
  const [selectedService, setSelectedService] = useState(null);
```

When user click on a service to order, call `setSelectedService(service)`
The condition to open modal now change from `showModal && <Modal />` to `!!selectedService && <Modal service ={selectedService}/>`

close modal by setSelectedService(null)